src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStation.cs

```diff
@@ -0,0 +1,323 @@
+        protected readonly IDownloadStationProxy _proxy;
+        protected readonly ISharedFolderResolverService _sharedFolderResolver;
+
+        public DownloadStation(IDownloadStationProxy proxy,
+                               ITorrentFileInfoReader torrentFileInfoReader,
+                               IHttpClient httpClient,
+                               IConfigService configService,
+                               IDiskProvider diskProvider,
+                               IRemotePathMappingService remotePathMappingService,
+                               Logger logger,
+                               ICacheManager cacheManager,
+                               ISharedFolderResolverService sharedFolderResolver)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            _proxy = proxy;
+            _sharedFolderResolver = sharedFolderResolver;
+            _sharedFolderResolver.Settings = Settings;
+        }
+
+        public override string Name => "Download Station";
+
+        protected virtual OsPath GetOutputPath(OsPath outputPath, DownloadStationTorrent torrent)
+        {
+            return outputPath + torrent.Title;
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            IEnumerable<DownloadStationTorrent> torrents;
+
+            try
+            {
+                torrents = _proxy.GetTorrents(Settings);
+            }
+            catch (DownloadClientException ex)
+            {
+                _logger.Error(ex);
+                return Enumerable.Empty<DownloadClientItem>();
+            }
+
+            var items = new List<DownloadClientItem>();
+
+            foreach (var torrent in torrents)
+            {
+                var outputPath = new OsPath($"/{torrent.Additional.Detail["destination"]}");
+
+                if (Settings.TvDirectory.IsNotNullOrWhiteSpace())
+                {                                                 
+                    if (!outputPath.Contains(new OsPath(Settings.TvDirectory)))
+                    {
+                        continue;
+                    }
+                }
+                else if (Settings.TvCategory.IsNotNullOrWhiteSpace())
+                {
+                    var directories = outputPath.FullPath.Split('\\', '/');
+                    if (!directories.Contains(Settings.TvCategory))
+                    {
+                        continue;
+                    }
+                }
+
+                var physicalPath = _sharedFolderResolver.ResolvePhysicalPath(outputPath.FullPath);
+
+                outputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, outputPath);
+                
+                var item = new DownloadClientItem()
+                {
+                    Category = Settings.TvCategory,
+                    DownloadClient = Definition.Name,
+                    DownloadId = torrent.Id,
```

Comment:
Prefix this with something. The DownloadId is used in History to find related history items and should be as globally unique as possible.
Imagine someone buying a new DiskStation, plugging it in and suddenly get weird behavior coz the Id starts at 1 again and newly added torrents start getting tracked completely wrong.
I couldn't find a reference of how 'Id' looks, whether it's just an ID or some Guid, if it's a Guid then disregard this comment, coz then it's already unique.